Libraries/LibWeb/Painting/BackgroundPainting.cpp

```diff
@@ -299,6 +299,9 @@ void paint_background(PaintContext& context, PaintableBox const& paintable_box,
             // Use a dedicated painting command for repeated images instead of recording a separate command for each instance
             // of a repeated background, so the painter has the opportunity to optimize the painting of repeated images.
             auto dest_rect = context.rounded_device_rect(image_rect);
+            if (dest_rect.is_empty()) {
+                dest_rect.set_size(1, 1);
+            }
```

Comment:
- I think only 1 dimension might be 0, in which case this shrinks the other dimension.
- It could do with some explanation either in a comment or in the commit message.

Reply:
Good points, I've split the check by dimension, added comments, and put the same check in the `for_each_image_device_rect` closure.